app/models/predicate.rb

```diff
@@ -17,4 +17,108 @@
 class Predicate < ApplicationRecord
   validates :uri, presence: true, uniqueness: true
   validates :pref_label, presence: true
+  after_create :assign_color, unless: proc { color.present? }
+
+  MAX_ALLOWED_WEIGHT = 5
+  MIN_ALLOWED_WEIGHT = 0
+
+  ###
+  # @description: Map of labels
+  ###
+  DEFAULT_LABELS = {
+    aggregated: "aggregated",
+    concept: "concept",
+    disaggregated: "disaggregated",
+    identical: "identical",
+    issue: "issue",
+    no_match: "no match",
+    not_applicable: "not applicable",
+    reworded: "reworded",
+    similar: "similar"
+  }.freeze
+
+  ###
+  # @description: Map of colors
+  ###
+  COLORS = {
+    dark_blue: "DarkBlue",
+    dark_green: "DarkGreen",
+    light_blue: "LightBlue",
+    light_green: "LimeGreen",
+    orange: "Orange",
+    red: "Red",
+    pink: "Pink",
+    unset: nil,
```

Comment:
Better use the `unset` value instead of `nil`, since it gets assigned on front end eventually anyway.